src/main/java/com/gojek/beast/util/DynamicMessageUtil.java

```diff
@@ -0,0 +1,33 @@
+package com.gojek.beast.util;
+
+import com.google.protobuf.DynamicMessage;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DynamicMessageUtil {
+    public static boolean isUnknownFieldExist(DynamicMessage dynamicMessage) {
+        if (dynamicMessage == null) {
+            return false;
+        }
+        List<DynamicMessage> dynamicMessageFields = new LinkedList<>();
```

Comment:
the root message should be there in the dynamicMessageFields

Reply:
it automatically add every DynamicMessage to accumulator when traversing the each of the DynamicMessage trees, including the very first node (root)
```
   private static void collectNestedFields(DynamicMessage root, List<DynamicMessage> accumulator) {
        List<DynamicMessage> nestedChild = root.getAllFields().values().stream()
                .filter(v -> v instanceof DynamicMessage)
                .map(o -> (DynamicMessage) o)
                .collect(Collectors.toList());

        nestedChild.forEach(m -> collectNestedFields(m, accumulator));
        accumulator.add(root);
    }
```